### PR TITLE
Add color picker example

### PR DESCRIPTION
**Pull request by alvinthen**
It's just a very simple example, here's how it looks like.

image

image

Related: #153

### CODE REVIEW

**Comment by nagad814** [comment]
Hi, Thank you for this commit. @thibaudcolas please can this merged. so it helps more people

**Comment by thibaudcolas** [comment]
A couple of years later… thank you @alvinthen, I’ll merge this now.

I ended up making one relatively big change – using inline styles for colors instead of entities. This is a relatively big trade-off:

- Entities are much simpler for this, but have the disadvantage that there can only be one entity applied to each chunk of text. For example this means you couldn’t use custom colors on links.
- Inline styles can be overlapped with entities – but they have no attached data, just an identifier.

So I ended up using dynamic inline style identifiers (for example `COLOR_FF0000`). This is relatively simple to implement although it doesn’t fit that well with Draftail’s APIs. Unfortunately, since inline styles can be stacked indefinitely, we also have to implement logic preventing multiple colors from being applied to the same range of text.

Long story short, I think this implementation works quite well, but it might not be appropriate for all use cases.
